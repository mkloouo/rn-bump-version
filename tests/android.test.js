const path = require("path");
const { versioner } = require("../lib/index");

const makeDefaultManager = ({
  semver,
  type = "minor",
  skipSemverFor = "ios",
  skipCodeFor = "ios",
  appName = "double",
  resetBuild = false,
} = {}) =>
  versioner(
    {
      root: path.join(__dirname, "android"),
      project: {
        android: {
          appName: appName,
          sourceDir: path.join(__dirname, "android"),
        },
      },
    },
    {
      type,
      semver,
      skipSemverFor,
      skipCodeFor,
      resetBuild,
    }
  );

test("successfully bump version", () => {
  const manager = makeDefaultManager().dryRun();

  expect(manager.buildGradle.content).toMatchSnapshot();
});

test("skip semVer when asked", () => {
  const manager = makeDefaultManager({ skipSemverFor: "all" }).dryRun();

  expect(manager.buildGradle.content).toMatchSnapshot();
});

test("preserve quotes style", () => {
  const manager = makeDefaultManager({ appName: "single" }).dryRun();

  expect(manager.buildGradle.content).toMatchSnapshot();
});

test("direct set semver string", () => {
  const manager = makeDefaultManager({ semver: "1.1.2" }).dryRun();

  expect(manager.buildGradle.content).toMatchSnapshot();
  expect(manager.packageJSON.content).toMatchSnapshot();
});

test("resets to 1 build number when changing semver", () => {
  const manager = makeDefaultManager({
    type: "patch",
    appName: "active",
    resetBuild: true,
  }).dryRun();

  expect(manager.buildGradle.content).toMatchSnapshot();
  expect(manager.packageJSON.content).toMatchSnapshot();
});

test("resets to 1 build number when skipping", () => {
  const manager = makeDefaultManager({
    appName: "active",
    resetBuild: true,
    skipSemverFor: "all",
  }).dryRun();

  expect(manager.buildGradle.content).toMatchSnapshot();
  expect(manager.packageJSON.content).toMatchSnapshot();
});
