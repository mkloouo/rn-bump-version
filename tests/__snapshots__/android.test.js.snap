// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`direct set semver string 1`] = `
"android {
    defaultConfig {
        versionCode 2
        versionName \\"1.1.2\\"
    }
}
"
`;

exports[`direct set semver string 2`] = `
Object {
  "version": "1.1.2",
}
`;

exports[`preserve quotes style 1`] = `
"android {
    defaultConfig {
        versionCode 2
        versionName '1.1.0'
    }
}
"
`;

exports[`resets to 1 build number when changing semver 1`] = `
"android {
    defaultConfig {
        versionCode 1
        versionName \\"1.0.1\\"
    }
}
"
`;

exports[`resets to 1 build number when changing semver 2`] = `
Object {
  "version": "1.0.1",
}
`;

exports[`resets to 1 build number when skipping 1`] = `
"android {
    defaultConfig {
        versionCode 1
        versionName \\"1.0.0\\"
    }
}
"
`;

exports[`resets to 1 build number when skipping 2`] = `
Object {
  "version": "1.0.0",
}
`;

exports[`skip semVer when asked 1`] = `
"android {
    defaultConfig {
        versionCode 2
        versionName \\"1.0.0\\"
    }
}
"
`;

exports[`successfully bump version 1`] = `
"android {
    defaultConfig {
        versionCode 2
        versionName \\"1.1.0\\"
    }
}
"
`;
